fxa/plugins/requests.py

```diff
@@ -33,24 +36,44 @@ class FxABrowserIDAuth(AuthBase):
 
     """
     def __init__(self, email, password, audience=None, with_client_state=False,
-                 server_url=core.DEFAULT_SERVER_URL):
+                 server_url=core.DEFAULT_SERVER_URL, cache=True,
+                 ttl=None):
         self.email = email
         self.password = password
         self.audience = audience
         self.with_client_state = with_client_state
         self.server_url = server_url
 
-    def __call__(self, request):
-        client = core.Client(server_url=self.server_url)
-        session = client.login(self.email, self.password, keys=True)
+        self.cache = cache
+        if self.cache is True:
+            if ttl is None:
+                ttl = DEFAULT_CACHE_EXPIRY - 1
+            self.cache = oauth.MemoryCache(ttl)
 
+    def __call__(self, request):
         if self.audience is None:
             url = urlparse(request.url)
             self.audience = "%s://%s/" % (url.scheme, url.netloc)
 
-        bid_assertion = session.get_identity_assertion(self.audience)
-        _, keyB = session.fetch_keys()
-        client_state = hexlify(sha256(keyB).digest()[0:16])
+        cache_key = sha256(''.join((
+            self.server_url, self.email,
+            self.password, self.audience)).encode('utf-8')).hexdigest()
```

Comment:
I suggest delimiting these with e.g. newlines to avoid any possibility of accidental collisions, i.e. `'\n'.join(...)`.  Also, it may make sense to factor this out into a shared helper function like:

```
def cache_key(*args):
  return sha256("\n".join(args).encode('utf-8')).hexdigest()
```